Apple Announcement

ryancdickson opened the pull request:
@lachellel @djpackham 

Please find this pull request for your review and (hopeful) approval.

If approved, this pull request will create a new announcement focused on sharing Apple-specific trust store guidance related to the removal of the FCPCA from Apple operating systems.

Comments:
- I'm happy to entertain any/all edits and recommendations. @clstmbrly helped provide editorial comments to ensure plain language and consistency with existing Playbook content/verbiage.
- I changed the file naming on the other announcements to better promote the Apple announcement. 
- Some of the FAQs included in the MSFT announcement were not applicable, so they were removed. Others were tweaked a bit due to better align with Apple.
- I verified the steps recommended (configuration profiles, terminal commands, keychain commands, etc.) worked on both macOS (my personal laptop) and iOS (my iPhone). Is it worthwhile to ask someone from NASA to review, given they have time and interest? 
- For completeness, we include reference to tvOS because it is impacted by COMMON's removal - but I am uncertain if Apple TV's are used across the USG. The configuration profile installation approach should work on tvOS, but I have not yet verified this yet (I have an Apple TV at home, so I will).  
- I think that's it!

Commit:
Create 04_apple_common_removal

Initial dump into GitHub

Commit:
Update 04_apple_common_removal

Commit:
Rename 04_apple_common_removal to 04_apple_common_removal.md

Commit:
Update 04_apple_common_removal.md

Minor updates

Commit:
Update 04_apple_common_removal.md

minor tweaks

Commit:
Rename 04_apple_common_removal.md to 01_apple_common_removal.md

Commit:
Rename 01_microsoft_constraint.md to 02_microsoft_constraint.md

Commit:
Rename 02_google_ct.md to 03_google_ct.md

Commit:
Rename 03_chrome_ballot_193.md to 04_chrome_ballot_193.md

Commit:
Update 01_apple_common_removal.md

Added terminal steps.

Commit:
Update 01_apple_common_removal.md

Commit:
Update 01_apple_common_removal.md

Minor tweaks.

Commit:
Update 01_apple_common_removal.md

Commit:
Update 01_apple_common_removal.md

Formatting tweaks.

Commit:
Update 01_apple_common_removal.md

Commit:
Update 01_apple_common_removal.md

Commit:
Update 01_apple_common_removal.md

Commit:
Update 01_apple_common_removal.md

Added configuration profile guidance. Rough cut.

Commit:
Update 01_apple_common_removal.md

Final edits for the day.

Commit:
Update 01_apple_common_removal.md

Rough cut at completing steps. Need to proof/verify.

Commit:
Update 01_apple_common_removal.md

Minor formatting tweaks for consistency.

Commit:
Update 01_apple_common_removal.md

clstmbrly committed:
Update 01_apple_common_removal.md

Editorial review (in process)

clstmbrly committed:
Update 01_apple_common_removal.md

Editorial review (in process)

clstmbrly committed:
Update 01_apple_common_removal.md

Editorial review (in process)

clstmbrly committed:
Update 01_apple_common_removal.md

Editorial review (in process)

clstmbrly committed:
Update 01_apple_common_removal.md

Format adjustment

clstmbrly committed:
Update 01_apple_common_removal.md

Format Fixes

Commit:
Update 01_apple_common_removal.md

Incorporating feedback from Celeste - round 1.

Commit:
Update 01_apple_common_removal.md

Final review (hopefully)

Commit:
Update 01_apple_common_removal.md

clstmbrly committed:
Update 01_apple_common_removal.md

Editorial updates

clstmbrly committed:
Update 01_apple_common_removal.md

Tweaks

clstmbrly committed:
Update 01_apple_common_removal.md

Editorial and flow

clstmbrly committed:
Update 01_apple_common_removal.md

Commit:
Update 01_apple_common_removal.md

Quick push to Federalist for review.

Commit:
Update 01_apple_common_removal.md

Final tweaks.

lachellel commented on _announcements/01_apple_common_removal.md:
```diff
@@ -0,0 +1,319 @@
+pubDate: June 8, 2018
+collection: announcements
+permalink: announcements/applepkichanges/
+description: Upcoming changes regarding Apple's Root Certificate Program could impact your agency. The Federal PKI Policy Authority has elected to remove our U.S. Government Root CA certificate (Federal Common Policy CA) from the Apple Operating System Trust Stores.  This change will impact government users of Apple iOS, macOS, and tvOS, starting in **September-October 2018**. <br><br> This chang ...
```
tvOS is accurate - but I can't think of a single use case for govt where this would be in use?  :+1:

lachellel commented on _announcements/01_apple_common_removal.md:
```diff
@@ -0,0 +1,319 @@
+| :-------- |:-------- |:-------- |
+| High Sierra (10.13) |	 iOS11 | tvOS11 |
+| Sierra (10.12) | iOS10 | tvOS10 |
+| El Capitan (10.11) | iOS9 | |
```
I don't think this is true?  it won't be retroactively applied to 10.11, 10.10 or 10.9 - or iOS7, iOS8, iOS9, and iOS10?

Check the apple notes - it's forward applied?

ryancdickson replied:
You're right - the note says "Removals are usually only forward - Removals can be backwards when there is a security concern via a revocation mechanism".

I mis-remembered that, or possibly confused it with the Symantec resolution, thanks for catching! I'll update to remove the older OS versions.

lachellel commented on _announcements/01_apple_common_removal.md:
```diff
@@ -0,0 +1,319 @@
+1. Logging into a government network with a PIV credential 
+2. Authenticating to a government Virtual Private Network (VPN) endpoint with a PIV credential
+3. Authenticating to an internet-facing, government collaboration portal with a PIV credential 
+4. Browsing with Safari, Chrome, or Edge/IE to a government **intranet** website that uses a Federal PKI CA-issued TLS/SSL certificate
```
Remove Edge/IE?  Can you even install Edge/IE on MacOS?

ryancdickson replied:
Edge is available on iOS. I'll make more clear.

lachellel commented on _announcements/01_apple_common_removal.md:
```diff
@@ -0,0 +1,319 @@
+Use any of these options to distribute your configuration profiles: 
+
+1. Via Apple's free _Configurator 2_ application with devices connected via USB 
+2. In an email message 
```
will email messages actually work? Won't the config files be stripped by email scanners?

ryancdickson replied:
GSA mail filter did not block the file, even with sender was outside the domain (verified).

lachellel commented on _announcements/01_apple_common_removal.md:
```diff
@@ -0,0 +1,319 @@
+## What Should I Do?
+
+These procedures are intended for Enterprise Administrators and/or Network Engineers. 
+
```
Can we add the sub-links to the sub-sections?  Download and Install are two separate sets of steps and options.   

I also can't tell if it's broken down by OS / device type, or actions. 

It's a bit confusing when sub-sub-sections are all "Option 1" "Option 2" etc.

ryancdickson replied:
I had tried this but ran into issues (links did not appear). I'll try again!

lachellel commented on _announcements/01_apple_common_removal.md:
```diff
@@ -0,0 +1,319 @@
+
+## Frequently Asked Questions
+
+### 1.&nbsp;&nbsp;Why is COMMON being removed? 
```
Remove the TLS/SSL

ryancdickson replied:
Removed.

lachellel commented on _announcements/01_apple_common_removal.md:
```diff
@@ -0,0 +1,319 @@
+### 1.&nbsp;&nbsp;Why is COMMON being removed? 
+The Federal PKI CAs don't comply with Apple's requirements for globally trusted TLS/SSL certificates. Apple's requirements include: 
+
+**a.&nbsp;&nbsp;Requirement for Fully-Qualified Domain Names (FQDNs)**<br>
```
Remove?  This isn't a driver for Apple's removal.

ryancdickson replied:
Removed.

lachellel commented on _announcements/01_apple_common_removal.md:
```diff
@@ -0,0 +1,319 @@
+**a.&nbsp;&nbsp;Requirement for Fully-Qualified Domain Names (FQDNs)**<br>
+Apple plans to restrict TLS/SSL certificates to only those certificates using FQDNs ending in .gov, .mil, or fed.us. Some federal agencies issue TLS/SSL certificates to intranet assets. These certificates either:&nbsp;&nbsp;don't have FQDNs; contain intranet domains that don't end in .gov, .mil, or fed.us; or use short names (aliases). Under Apple's requirements, these agencies would need to reis ...
+
+**b.&nbsp;&nbsp;Requirement for public audit**<br>
```
Update or remove?  Apple requires webtrust audits, and public documents.  It's also not related to the .gov / .mil domain constraint.

ryancdickson replied:
Removed.

lachellel commented on _announcements/01_apple_common_removal.md:
```diff
@@ -0,0 +1,319 @@
+**c.&nbsp;&nbsp;Requirement to disclose Certificate Practice Statements and Incident Post-Mortem Reports**<br>
+Public trust requires public disclosure and transparency. All Federal PKI CAs would be required to publicly post their Certificate Practice Statements and their Audit Letters. The Federal PKI community has attempted to disclose all Certificate Practice Statements for a number of years. However, some federal agencies include sensitive information in these documents and cannot disclose the document ...
+
+**d.&nbsp;&nbsp;Requirement to create new issuing Certification Authorities (CAs)**<br>
```
I'd remove? 

If we want to be specific here - the four items are: 

Certificate Transparency
Public disclosure and audits
Separating CAs by purpose
Symantec Distrust

Three of the four could be listed as informational?

ryancdickson replied:
For now, I opted to remove. 

I'm struggling to think of a straight forward way of tying in CT/Symantec Distrust without really adding some weight/complexity to the page (e.g., to explain the CT influence well I feel like I'd need to reference root_stores.json and describe how user agents are tracked/pruned over time to highlight how removing now will save us pain in the future). 

What do you think?

lachellel replied:
Makes sense.  I was thinking of the upcoming CT, not just the chrome implementation too.  You are correct IMO: let's keep it simpler first.  It's already long enough!

lachellel commented on _announcements/01_apple_common_removal.md:
```diff
@@ -0,0 +1,319 @@
+
+You can run a report on all issued certificates or, if your agency has an agreement with a Federal PKI Shared Service Provider (SSP), you can request that the SSP run the report.
+
+You can scan your intranet websites in coordination with your CISO teams. There are existing tools to use, or you can use the DHS NCATS "**pshtt**" tool, which will also check for cipher suites and mis-configurations on the intranet websites:
```
Remove pshtt

ryancdickson replied:
Removed.

Commit:
Update 01_apple_common_removal.md

Updates per feedback.

clstmbrly committed:
Update 01_apple_common_removal.md

Adding sub-links

clstmbrly committed:
Update 01_apple_common_removal.md

Adding sub-links

clstmbrly committed:
Update 01_apple_common_removal.md

Adding sub-links

lachellel committed:
option for consolidating

Changes included consolidating the download and install options together (for both MacOS and iOS), adding in more sub-links, and removing "COMMON" from all the sub-section headers.

lachellel committed:
links

fixing the links

lachellel committed:
more updates

lachellel committed:
Merge pull request #238 from GSA/lachellel-patch-1

option for consolidating

lachellel commented:
@ryancdickson 

PR #238 merged some changes.  

In addition, I removed live links to the root cert.   We don't want - as a general practice - someone clicking on the live link and downloading a root certificate.  We did the same in the piv guides - the URL is shown but not as an active link to click.

lachellel commented:
@ryancdickson @clstmbrly let me know how the changes look, and I'll merge.

clstmbrly commented:
@lachellel I'm reviewing.  Thanks

Commit:
Update 01_apple_common_removal.md

Minor tweak.

Commit:
Update 01_apple_common_removal.md

Typo correction.

Commit:
Update 01_apple_common_removal.md

Correcting branding on macOS.

Commit:
Update 01_apple_common_removal.md

Typo correction.
